druid-derive/src/lens.rs

```diff
@@ -86,69 +70,26 @@ fn derive_struct(input: &syn::DeriveInput) -> Result<proc_macro2::TokenStream, s
         }
     });
 
+    let associated_items = fields.iter().map(|f| {
+        let field_name = &f.ident;
+        quote! {
+            pub const #field_name: #field_name = #field_name;
```

Comment:
If this is only ever accessed via the associated const, we're less concerned about having the actual name of the struct match the name of the field. I'm thinking about the possible hygene concerns about having a struct named, sat, `value` or `inner` or similar just sort of hanging out in the top level of a module? one option could be to create a per-type module to stash these in, so that you have something like,

```rust
//
#[derive(Lens)]
struct MyData {
    value: usize,
    name: String,
}

pub mod my_data_lenses {
    pub struct name;
    pub struct value;
}

impl MyData {
    const name: super::my_data_lenses::name = super::my_data_lenses::name;
}
```

again not sure this is necessary, just thinking out loud.

Reply:
That's actually exactly what this does! Rolling back to the original commit for now, though, since it's an unambiguous win. Will open a new PR for further discussion.

Reply:
okay this comment may be misguided, ignore as needed :)